vaadin-spreadsheet-flow-parent/vaadin-spreadsheet-flow/src/main/resources/META-INF/resources/frontend/vaadin-spreadsheet/vaadin-spreadsheet.js

```diff
@@ -196,15 +196,31 @@ export class VaadinSpreadsheet extends LitElement {
     super.connectedCallback()
     this.observer && this.observer.observe(this);
     // Restore styles in the case widget is reattached, it happens e.g in client router
-    this.styles && this.styles.forEach(e => document.head.appendChild(e));
+    let i = 0;
+    this.styles && this.styles.forEach(e => {
+      let addedNode = document.head.appendChild(e);
```

Comment:
No need to redefine a reference to the same node (`e` and `addedNode` here refer to the same node)